lib/matplotlib/widgets.py

```diff
@@ -574,38 +574,72 @@ def _clicked(self, event):
                 self.set_active(i)
                 break
 
-    def set_active(self, index):
+    def set_active(self, index, state=None):
```

Comment:
I see your concern here, and we should try limit them, sure. However, in this particular case, presenting a `set_active(index, *args)` format to user is confusing as users are implied to provide a `list`, to later only enforce a single value to it. I think its much more readable use `status`. But I'm open to discussion here, and learn from other developers too.

Reply:
Clarifying this is what the docstring is for.

Reply:
I'd like to add the whole interface needs a reworking (remove tricky behaviour of set_active, and add methods for toggle, set and unset etc). if there would be an opportunity to implement any backwards incompatible changes in the future, time is better invested in rewriting the interfaces than adding these tricky inputs/documentation at this point. Perhaps I lack the necessary perspective (esp. from the matplotlib maintainer/core-dev experience), but this is just my 2 cents.

Reply:
What about `def set_active(self, index, state='toggle'):`?

Reply:
In my view that also runs foul of "can be interpreted as a bool" (and if anything it's worse than using None for that, as other strings will be considered True(?) or excluded(?), whereas at least None is somewhat special).

Reply:
Concerning reworking the interface: I'm not very deep into the widgets stuff, but from what I've seen I'd at least consider a complete rewrite rather than introducing significant backward-incompatible changes. Basically, the widgets are not essential to Matplotlib and one could create an independent 3rd party package. Note however, that getting GUI widgets right (both the interfaces and the mechanics below) is not a piece of cake and I'm not advocating taking on the endeavor. Anybody willing to do so should be proficient in at least one (better two) of the existing major GUI frameworks.

Reply:
Ok, last try :smile: 
~~~
 _TOGGLE = object()
def set_active(self, index, state=_TOGGLE):
~~~
Otherwise, I think *None* would be ok.

Reply:
that's what I think too - current None implementation (for all its technical appearances) looks readable to me! If this is okay with you guys, is it ready to merge?

Reply:
Actually using a sentinel value looks good to me.

Reply:
you mean `state='toggle'` by default, and allowing for `True` or `False` as valid values for `state`?

Reply:
Let me know what else would be needed to approve this - it will be nice to close this PR this year! :)

Reply:
`state=_TOGGLE` as above.